omegaconf/dictconfig.py

```diff
@@ -170,7 +171,7 @@ def _validate_set(self, key: Any, value: Any) -> None:
         if vk == ValueKind.INTERPOLATION:
             return
         self._validate_non_optional(key, value)
-        if (isinstance(value, str) and value == "???") or value is None:
+        if vk is ValueKind.MANDATORY_MISSING or value is None:
```

Comment:
Minor: enums are usually compared with `==` (since it's possible to be equal with a different object)

Reply:
sure.